tools/clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -2559,7 +2559,8 @@ void ScalarExprEmitter::EmitUndefinedBehaviorIntegerDivAndRemCheck(
 
     llvm::Value *IntMin =
       Builder.getInt(llvm::APInt::getSignedMinValue(Ty->getBitWidth()));
-    llvm::Value *NegOne = llvm::ConstantInt::get(Ty, -1ULL);
+    llvm::Value *NegOne =
+        llvm::ConstantInt::get(Ty, std::numeric_limits<int64_t>::min());
```

Comment:
Shouldn't this be uint64_t?

Reply:
Should be std::numeric_limits<uint64_t>::max() as well.